src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -30,17 +30,19 @@ internal sealed partial class SourceNamespaceSymbol : NamespaceSymbol
         private ImmutableArray<Symbol> _lazyAllMembers;
         private ImmutableArray<NamedTypeSymbol> _lazyTypeMembersUnordered;
 
+        private readonly object _aliasesAndUsingsLock = new();
```

Comment:
>_aliasesAndUsingsLock [](http://example.com/codeflow?start=32&length=21)

It looks like ```SegmentedDictionary``` is a class and both instances are stored in private fields. I don't think we need an extra object for locking.
Also, instead of trying to add each item to a pre-allocated dictionary on demand and struggling with synchronization, let's try to start with a null dictionaries, allocate and populate the dictionaries completely on first access (together and in a manner the dictionary used to be populated before the recent change), then do an interlocked compare exchange for both.